esdt-safe/src/lib.rs

```diff
@@ -687,4 +725,7 @@ pub trait EsdtSafe:
         address: &ManagedAddress,
         token_id: &TokenIdentifier,
     ) -> SingleValueMapper<BigUint>;
+
+    #[storage_mapper("failedRefunds")]
+    fn failed_refunds(&self) -> VecMapper<Transaction<Self::Api>>;
```

Comment:
where are we processing these failed refunds?

Reply:
Added a `add_refund_batch_for_failed_tx` endpoint to process the failed_refunds.